x-pack/plugins/security_solution/docs/testing/test_plans/detection_response/rule_management/coverage_overview_dashboard.md

```diff
@@ -0,0 +1,191 @@
+And color tiles according to filters and dashboard legend
+```
+
+#### **Scenario: User clicks on tile**
+
+**Automation**: 1 e2e test + 1 unit test.
+
+```Gherkin
+Given rules installed/created
+And rules enabled
+When user navigates to Coverage Overview page
+And clicks on technique tile with X rules
+Then the popover should display X rule names under their corresponding rule activity secition
+And each name should link to its own rule details page
+```
+
+### Coverage overview workflow: filters
+
+#### **Scenario: No filters are present**
+
+**Automation**: 1 integration test.
+
+```Gherkin
+Given coverage overview page is loaded with rule data
+When no filters or search term is present
+Then all rule data is fetched and rendered
+```
+
+#### **Scenario: Users enables filters**
+
+**Automation**: integration tests + e2e tests.
+
+```Gherkin
+Given coverage overview page is loaded with rule data
+When <type> filter(s) is/are enabled
+Then all <type> rule data is fetched and rendered
+
+Examples:
+  | type                   |
+  | enabled                |
+  | disabled               |
+  | prebuilt               |
+  | custom                 |
+  | enabled and disabled   |
+  | prebuilt and custom    |
+  | all                    |
+```
+
+#### **Scenario: Search term filter present**
+
+**Automation**: 1 integration test + 1 e2e test.
+
+```Gherkin
+Given coverage overview page is loaded with rule data
+When search term filter is present in search box
+And user submits the search
+Then only search-filtered rule data is fetched and rendered
+```
+
+### Coverage overview workflow: rule enabling
+
+#### **Scenario: User can't enable all rules when no disabled rules**
+
+**Automation**: 1 unit test.
+
+```Gherkin
+Given coverage overview page is loaded with rule data
+When user clicks on a technique tile with no disabled rules
+Then "enable all disabled" button should be disabled
+```
+
+#### **Scenario: User enables all rules for technique**
+
+**Automation**: 1 e2e test.
+
+```Gherkin
+Given coverage overview page is loaded with rule data
+When user clicks on a technique tile with X disabled rules
+And clicks "enable all disabled" button
+Then all X disabled rules hould be enabled
+And user should see success toast message for X rules enabled
+And page should update data
+```
+
+#### **Scenario: User can't enable rules when they don't have CRUD privileges**
+
+**Automation**: 1 unit test.
+
+```Gherkin
+Given coverage overview page is loaded with rule data
+And user that doesn't have CRUD permissions
+When user clicks on a technique tile with disabled rules
+Then "enable all disabled" button should be disabled
+```
+
+### API specific cases
+
+#### **Scenario: Non-security rules are present**
+
+**Automation**: 1 integration test.
+
+```Gherkin
+Given non-security rules are present in instance
+When coverage overview data is fetched
+Then only security rule data is returned
+And no error is thrown
+```
+
+#### **Scenario: Invalid API body**
```

Comment:
We don't validate the body 😅 Just check the return code. It looks like a CASE for `Error is handled when API error is returned` scenario.